### DISCUSSION TITLE
How to add upstream/topic_branch to local, forked repo

### DISCUSSION THREAD

**Discussion by max123kl** [community/community#23033] [New to GitHub]
<p>The following scenario is given:</p>
<ol>
<li>upstream maintainer creates topic branch “ABC_branch”</li>
<li>my fork doesn’t contain this new branch yet (neither locally nor on GitHub)</li>
</ol>
<p>How can I add (clone or download, what ever) the new ABC_branch to my local repository without touching my local master or any other existing branch?</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/max123kl">@max123kl</a>,</p>
<p>Thank you for being here! <code>git rebase</code> in its simplest form is a command which will “port” another branch (Branch A) into the branch where you are currently working (Branch B), by applying all of your unique commits from Branch B on top of Branch A and replacing Branch B with this revised version. That way, you can “catch up” to other branches by re-writing git history in your local git.</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>If you have a remote in your local repository named <code>upstream</code> that points to their repository and a remote named <code>origin</code> that points to your GitHub repository, you can do the following:</p>
<pre><code># Creates a local branch named `ABC_branch`
git checkout -b ABC_branch

# Fetches the `ABC_branch` content from `upstream` and merges it into your local branch
git pull upstream ABC_branch
</code></pre>
<p>Then, if you want to have <code>ABC_branch</code> in your fork of their repository, you can use:</p>
<pre><code>git push origin ABC_branch
</code></pre>
<p>I hope that helps!</p>
